### PR TITLE
feat: YAML key delimiter for dot notation in stack files

### PR DESCRIPTION
**Pull request by osterman**
## what

- Implement configurable `settings.yaml.key_delimiter` that expands dotted YAML keys into nested maps in stack files (e.g., `metadata.component: vpc-base` -> `metadata: { component: vpc-base }`)
- Unquoted keys containing the delimiter expand automatically; quoted keys stay literal (escape mechanism)
- Support custom delimiters, not just `.` (e.g., `::`)
- Mark feature as experimental with runtime notifications via `ui.Experimental()`
- 21 comprehensive unit tests covering expansion logic, merging, conflicts, quoted preservation, edge cases

## example

With `settings.yaml.key_delimiter` enabled:

```yaml
settings:
  yaml:
    key_delimiter: "."
```

Stack files can use concise unquoted dotted keys for nested component configuration. `metadata` stays at the component level; it is not nested under `vars`:

```yaml
components:
  terraform:
    vpc:
      metadata.component: vpc-base
      metadata.description: Base VPC component
      vars.tags.environment: prod
      vars.tags.team: network
```

Atmos expands that as if the stack had been written in nested form:

```yaml
components:
  terraform:
    vpc:
      metadata:
        component: vpc-base
        description: Base VPC component
      vars:
        tags:
          environment: prod
          team: network
```

Quoted mapping keys remain literal, so existing dotted keys can be preserved:

```yaml
components:
  terraform:
    app:
      vars:
        "output.json": true
        'config.file': /etc/app.conf
```

## why

Documentation claimed dot notation worked in stack files but it didn't. Go's `yaml.v3` treats dotted keys literally without expansion. Users want concise notation instead of deeply nested YAML structures. This brings feature parity with `atmos.yaml`'s native support via Viper's `deepSearch()` algorithm. Backwards compatible (disabled by default) and opt-in.

## references

Extends experimental features system (like `atmos devcontainer`, `atmos list affected`). Respects `settings.experimental` mode configuration (silence/warn/error/disable).

## Summary by CodeRabbit

* **New Features**
  * Experimental YAML key expansion (`settings.yaml.key_delimiter`): unquoted dotted keys can expand into nested maps in stack and settings files; custom delimiters are supported.
  * Enhanced experimental configuration checks so YAML/settings features follow the same warn/disable/error modes.
* **Documentation**
  * Added a blog post and expanded YAML dot-notation reference with examples for quoting, custom delimiters, and edge cases.
* **Tests**
  * Added coverage for delimiter expansion behavior and merge/conflict scenarios.
* **Chores**
  * Updated the Playwright-Go license reference.

### CODE REVIEW

File: pkg/datafetcher/schema/atmos/manifest/1.0.json [hunk 1]
```diff
@@ -613,6 +613,17 @@
...
+              "type": "object",
+              "description": "YAML parsing settings",
+              "additionalProperties": false,
+              "properties": {
+                "key_delimiter": {
+                  "type": "string",
+                  "description": "Delimiter for expanding dotted keys into nested maps (e.g., '.' or '::'). Empty or omitted disables expansion."
+                }
+              }
             }
```
[the remark this answers is not in the document]
Comment (osterman) [reply pkg/datafetcher/schema/atmos/manifest/1.0.json]: Good catch! Removed `additionalProperties: false` from the `yaml` object in both `pkg/datafetcher/schema/atmos/manifest/1.0.json` and `pkg/datafetcher/schema/config/global/1.0.json`. The Go struct already controls what is parsed, so the schema does not need to be the enforcement point.

**Review by aknysh — APPROVED** [review]
